src/Bicep.Decompiler/TemplateConverter.cs

```diff
@@ -535,23 +553,60 @@ private ObjectSyntax ParseJObject(JObject jObject)
             var properties = new List<ObjectPropertySyntax>();
             foreach (var (key, value) in jObject)
             {
-                ObjectPropertySyntax objectProperty;
-                if (ExpressionsEngine.IsLanguageExpression(key))
+                if (key == "copy" && value is JArray)
+                {
+                    foreach (var entry in value)
+                    {
+                        if (entry is not JObject copyProperty)
+                        {
+                            throw new ConversionFailedException($"Expected a copy object", entry);
+                        }
+
+                        var name = TemplateHelpers.AssertRequiredProperty(copyProperty, "name", "The copy object is missing a \"name\" property").ToString();
+                        var count = TemplateHelpers.AssertRequiredProperty(copyProperty, "count", "The copy object is missing a \"count\" property");
+                        var input = TemplateHelpers.AssertRequiredProperty(copyProperty, "input", "The copy object is missing a \"input\" property");
+
+                        var objectProperty = PerformScopedAction(() => {
+                            input = ExpressionHelpers.ReplaceFunctionExpressions(input, function => {
+                                if (!StringComparer.OrdinalIgnoreCase.Equals(function.Function, "copyIndex"))
+                                {
+                                    return;
+                                }
+
+                                if (function.Parameters.Length == 1 && function.Parameters[0] is JTokenExpression copyIndexName && copyIndexName.Value.ToString() == name)
```

Comment:
>function.Parameters.Length == 1 [](start = 36, length = 31)

The copyIndex() function has an optional offset parameter as well.

Reply:
I see we handled it for resource copy also. Should probably either block it or handle it because otherwise, the template intent will change during decompilation.

---
In reply to: [587243171](https://github.com/Azure/bicep/pull/1640#discussion_r587243171) [](ancestors = 587243171)

Reply:
Added handling for it